src/com/climate/geojson_schema/core.clj

```diff
@@ -125,7 +126,9 @@
          {:features [Feature]
           :type (eq "FeatureCollection")}))
 
-(def GeoJSON (either Geometry
-                     GeometryCollection
-                     Feature
-                     FeatureCollection))
+(def GeoJSON
+  (named (conditional #(contains? #{"Point" "MultiPoint" "LineString" "MultiLineString" "Polygon" "MultiPolygon"} (:type %)) Geometry
```

Comment:
Not sure if contains? is required here, clojure.set itself is a first class function that checks if the elements is in the set : 

```
user=> (#{1 2} 1)
1
user=> (#{1 2} 4)
nil
```